File: OneDimension.java
package physics.mechanics.motion.generic;

import physics.mechanics.object.Object;

public class OneDimension {
	
	private Object object;
	private double distance, acc, iVelocity;
	private double force;
	
	public void initiate(Object o, double u) {
		this.object = o;
		this.iVelocity = u;
	}
	
	public double calculateAcceleration(double v2, double t) {
		acc = (v2-iVelocity)/t;
		return acc;
	}
	
	public double calculateDistance(double v2, float t) {
		if(acc==0d)
			calculateAcceleration(v2, t);
		distance = iVelocity*t + (double)1/2*acc*t*t;
		return distance;
	}
	
	public double calculateForce(double v2, double t) {
		if(acc==0d)
			calculateAcceleration(v2, t);
		force = object.getMass()*acc;
		return force;
	}
	
	public String getAccelerationEquation() {
		String s = String.format("a(t) = (v - %.2d)/t", iVelocity);
		return s;
	}
	
	public String getDistanceEquation() {
		String s = String.format("s(t) = %.2d*t + %.2d*t^2", iVelocity,acc/2);
		return s;
	}

}

File: python/physics/mechanics/motion/generic/OneDimension.py
import numpy as nm
import matplotlib.pyplot as plt

def calculateAcceleration(iVelocity,v2,t):
    return float(v2-iVelocity)/t

def calculateDistance(iVelocity,v2,t,acc=0):
    if acc == 0:
        acc = calculateAcceleration(iVelocity,v2,t)
    return float(iVelocity)*t + float(acc)*t*t/2

def calculateForce(mass,iVelocity,v2,t,acc=0):
    if acc == 0:
        acc = calculateAcceleration(iVelocity,v2,t)
    return mass*acc

def getDistanceEquation(t, iVelocity, acc=0):
    if acc == 0:
        acc = calculateAcceleration(iVelocity,v2,t)
    return float(iVelocity)*t + float(acc)*(t*t)/2

#
def initiate():
    mass = input("Enter the mass of the object: ")
    iVelocity = input("Enter the initial velocity of the object: ")
    v2 = input("Enter the current velocity of the object: ")
    t = input("Enter the duration of motion: ")
    calculate(mass,iVelocity,v2,t)

def calculate(mass,iVelocity,v2,t):
    acc = calculateAcceleration(iVelocity,v2,t)
    print("Acceleration is "+str(acc)+" m/s2")
    distance = calculateDistance(iVelocity,v2,t,acc)
    print("Distance travelled is "+str(distance)+" m")
    force = calculateForce(mass,iVelocity,v2,t,acc)
    print("Internal force is "+str(force)+" N")
    plot_graphs(mass,iVelocity,v2,t,acc)

def plot_graphs(mass,iVelocity,v2,t,acc):
    t = nm.arange(0.0,10.0,0.1)
    plt.figure(1)
    plt.subplot(111)
    plt.plot(t,getDistanceEquation(t, iVelocity, acc),'r--')
    plt.show()

#
initiate()
